src/web_scraper/web_scraper.py

```diff
@@ -0,0 +1,125 @@
+
+
+"""
+
+from docx import Document
+from supabase import create_client, Client
+from datetime import datetime
+import feedparser
+import os
+import re
+
+senate_rss_feed = "https://www.legis.state.pa.us/WU01/LI/RSS/SenateBills.xml"
+house_rss_feed = "https://www.legis.state.pa.us/WU01/LI/RSS/HouseBills.xml"
+
+sb_api_url = "https://vsumrxhpkzegrktbtcui.supabase.co" #os.environ["API_URL"] 
+sb_api_key = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6InZzdW1yeGhwa3plZ3JrdGJ0Y3VpIiwicm9sZSI6ImFub24iLCJpYXQiOjE2ODQ3OTU2NzUsImV4cCI6MjAwMDM3MTY3NX0.9lafalZT9FJW1D8DAuIMrsRX0Gs6204nV8ETfGslrqI" #os.environ["API_KEY"] 
+
+def extract_and_upload_senate_bill_metadata():
+    sen_rss = feedparser.parse(senate_rss_feed)
+    supabase: Client = create_client(sb_api_url, sb_api_key)
+    for bill in sen_rss.entries:
+        #print("checking bill ", bill.title)
+        if bill is None: continue
+
+        bill_no_regex = '(Senate Bill \d*|Senate Resolution \d*)'
+        bill_no = re.findall(bill_no_regex, bill.title)[0].split(' ')[-1]
+        
+        printer_no_regex="Printer's Number \d*"
+        printer_no = re.findall(printer_no_regex, bill.title)[0].split(' ')[-1]
+
+        #OLD CODE that doesn't work
+        #title_elements = bill.title.split(' ') #grabbing bill and printer numbers from the title (not explicitly given in RSS feed)
+        #bill_no = "".join(char for char in title_elements[2] if char.isdigit()) #ensuring that only the actual digits of the bill number are extracted
+        #printer_no = "".join(char for char in title_elements[5] if char.isdigit()) #ensuring that only the actual digits of the printer number are extracted
+
+        #change if bill["parss_enacted"] == "YES" and title_elements[1] == "Bill": #make sure the bill was enacted and it's a bill instead of a resolution        
+        bill_existing_records = supabase.table('Revisions').select("*", count="exact").eq("bill_id", bill_no).eq("printer_no", printer_no).eq("legislative_body", "senate").execute()
+        if bill_existing_records.count == 0: #if the bill hasn't already been recorded in the supabase table, then post it (so you don't upload duplicates)
+            response = supabase.table('Revisions').insert( #why is there no python docs for upsert?
+                {
+                    # "created_at": metadata["created_at"], #supabase already defaults to now(), so this is kind of unnecessary for now
+                    "bill_id": bill_no,
+                    "printer_no": printer_no, 
+                    "full_text_link": bill["link"], 
+                    "publication_date": bill["published"], 
+                    "legislative_body": "senate"
+                    # "active_summary_id": metadata["active_summary_id"], #leaving out foreign keys for now
+                }).execute()
```

Comment:
This is a stylistic recommendation and can thus be considered optional, but my personal preference is to minimize the number of indentations as this creates confusing "arrow code" -- at a glance, a developer can't always read the white space and know what if clause they're embedded in, which can lead to bugs down the line. 

[This article](https://blog.codinghorror.com/flattening-arrow-code/) does a decent job of discussing how to get rid of it, but my two recommendations here would be:

1. See if you can replace if-conditionals with guard clauses (i.e. you just exit the function if the condition is NOT met)
2. Decompose code body into separate functions. This also helps with preventing having different levels of abstraction be in the same function.